feat: Add CustomArrayList and CustomHashSet implementations

tomrist9 opened the pull request:
Pull Request Description
This pull request introduces the following features:

CustomArrayList
A custom implementation of ArrayList with the following functionalities:
Add (put) elements to the list.
Retrieve elements by index using the get method.
Delete elements by index using the delete method.
Automatically resizes when the internal capacity is exceeded.
Includes an overridden toString() method for better debugging and output representation.
CustomHashSet
A custom implementation of HashSet that ensures object uniqueness and provides:
Add (put) unique elements.
Check if an element exists using the contains method.
Delete elements using the delete method.
Iterate over elements using a custom iterator and support for while and for-each loops.
Automatic resizing when the load factor exceeds the predefined threshold.
Main Class
Added a Main class to test the implemented functionalities:
Validates the behavior of CustomArrayList and CustomHashSet.
Demonstrates adding, retrieving, deleting, resizing, and iterating elements.

tomrist9 committed:
fix(ticket-deserialization): handle JSON deserialization errors for Ticket class

- Added missing 	icketClass field in Ticket class to match JSON structure.
- Implemented no-argument constructor in Ticket class to enable Jackson deserialization.
- Configured Jackson to ignore unknown JSON fields using @JsonIgnoreProperties annotation.
- Updated loadTicketsFromResources method to support JSON arrays and single objects.

These changes ensure robust handling of JSON deserialization and prevent errors from unrecognized fields or missing constructors.

tomrist9 committed:
feat: Implement CustomArrayList and CustomHashSet with required methods

- Added CustomArrayList class with the following features:
  - Add (put) elements
  - Get elements by index
  - Delete elements by index
  - Automatic resizing when capacity is exceeded
  - Overridden toString() method for better output representation

- Added CustomHashSet class with the following features:
  - Add (put) unique elements
  - Check element existence (contains)
  - Delete elements
  - Iterate through elements using for-each or while loops
  - Automatic resizing based on load factor to maintain efficiency

- Wrote a Main class to test all methods, ensuring correctness for both classes.

tomrist9 committed:
changes made in delete method

tomrist9 committed:
changes made in main class and customhashset class

tomrist9 committed:
last changes have been added

tomrist9 committed:
Update CustomHashSet.java

Darya-Drutsko commented on andersen-homework/src/main/java/org/util/CustomArrayList.java:
```diff
@@ -0,0 +1,47 @@
+
+public class CustomArrayList <T>{
+    private Object[] items;
+    int size=0;
```
What this field has default access modifier?

tomrist9 replied:
int size has a default access modifier.It makes field accessible within the same package

Darya-Drutsko commented on andersen-homework/src/main/java/org/util/CustomArrayList.java:
```diff
@@ -0,0 +1,47 @@
+    private Object[] items;
+    int size=0;
+    public CustomArrayList() {
+        items = new Object[10];
```
put please 10 as a constant

tomrist9 replied:
done

Darya-Drutsko replied:
Don't see the result

tomrist9 replied:
done

Darya-Drutsko commented on andersen-homework/src/main/java/org/util/CustomArrayList.java:
```diff
@@ -0,0 +1,47 @@
+package org.util;
```
need to apply formatting for the class

Darya-Drutsko replied:
Don't see the result

tomrist9 replied:
done

Darya-Drutsko commented on andersen-homework/src/main/java/org/util/CustomArrayList.java:
```diff
@@ -0,0 +1,47 @@
+    }
+
+
+    public int getSize() {
```
Why do we need 2 methods with the same body?

tomrist9 replied:
I used getter firstly, but could not call it in main class. so created size class manually. But then I removed getter annotation and created getter manually.

tomrist9 replied:
please review new commit after i made changes

Darya-Drutsko commented on andersen-homework/src/main/java/org/util/CustomHashSet.java:
```diff
@@ -0,0 +1,93 @@
+
+import java.util.LinkedList;
+
+public class CustomHashSet<T> {
```
Cool implementation!

Darya-Drutsko reviewed (commented):
No need to delete previous ones files, lets have them separate(e.g. in different packages), but in one pr

tomrist9 committed:
style: Apply consistent formatting and adhere to Java coding conventions

- Adjusted indentation and spacing for improved readability
- Organized imports and removed unused ones
- Added proper line breaks and consistent braces
- Ensured fields and methods follow naming conventions
- Improved documentation with Javadoc comments for clarity

tomrist9 committed:
put 10 as a constant and applied formatting to the class

tomrist9 committed:
put 10 as a constant and applied formatting to the class

Darya-Drutsko reviewed (commented):
Always gitignore .idea folder and return deleted files, they should still be present in develop branch after merging

tomrist9 committed:
Add .idea to .gitignore

tomrist9 committed:
Stop tracking .idea folder and add to .gitignore

tomrist9 committed:
ignored idea
